types/model/name.go

```diff
@@ -298,6 +298,13 @@ func (n Name) LogValue() slog.Value {
 	return slog.StringValue(n.String())
 }
 
+func (n Name) EqualFold(o Name) bool {
```

Comment:
This was introduced during an iteration of this PR, but then become unused by the PR. I think it's fine to leave in, since it can (should) be used elsewhere. If more changes need to be made, I'll remove it then if anyone has a strong argument to remove it.